docs/book/v3/quick-start.md

```diff
@@ -112,14 +190,58 @@ $array  = $delegating->extract(new Artist());
 $artist = $delegating->hydrate($data, new Artist());
 ```
 
-### Laminas\\Hydrator\\ObjectPropertyHydrator
+<a name="base_interfaces"></a>
+## Base Interfaces
+
+### ExtractionInterface
+
+```php
+<?php
+
+namespace Laminas\Hydrator;
+
+interface ExtractionInterface
+{
+    /**
+     * Extract values from an object
+     *
+     * @return mixed[]
+     */
+    public function extract(object $object) : array;
+}
+```
+
+### HydrationInterface
+
+```php
+<?php
+
+namespace Laminas\Hydrator;
+
+interface HydrationInterface
+{
+    /**
+     * Hydrate $object with the provided $data.
+     *
+     * @param mixed[] $data
+     * @return object The implementation should return an object of any type.
+     *     By purposely omitting the return type from the signature,
+     *     implementations may choose to specify a more specific type.
+     */
+    public function hydrate(array $data, object $object);
+}
+```
+
+### HydratorInterface
 
-Any data key matching a publicly accessible property will be hydrated; any public properties
-will be used for extraction.
+```php
+<?php
 
-### Laminas\\Hydrator\\ReflectionHydrator
+namespace Laminas\Hydrator;
 
-Similar to the `ObjectPropertyHydrator`, but uses [PHP's reflection API](http://php.net/manual/en/intro.reflection.php)
-to hydrate or extract properties of any visibility. Any data key matching an
-existing property will be hydrated; any existing properties will be used for
-extraction.
+interface HydratorInterface extends
+    ExtractionInterface,
+    HydrationInterface
+{
+}
+```
```

Comment:
Should be extracted to a separate page which describes the creation of custom hydrator otherwise the interfaces in the documentation are not useful.

Reply:
That's going to take a bit of time to complete, but I'll get it done.

Reply:
I'm going to do that in a separate PR.